src/test/java/com/github/tomakehurst/wiremock/MultipartBodyMatchingAcceptanceTest.java

```diff
@@ -250,4 +250,70 @@ void acceptsAMultipartRequestWithCamelcasedContentTypeInformationPrefixedWithSpa
     }
     assertThat(connection.getResponseCode(), is(200));
   }
+
+  @Test
+  void acceptsAMultipartRelatedSOAPWithAttachmentRequest() throws Exception {
+    final String soapBody =
+        "<soap:Envelope xmlns:soap=\"http://www.w3.org/2003/05/soap-envelope\">\r\n"
+            + "  <soap:Header></soap:Header>\r\n"
+            + "  <soap:Body>\r\n"
+            + "    <ns1:Test xmlns:ns1=\"http://www.test.org/some-test-namespace\">\r\n"
+            + "      <ns1:Attachment>\r\n"
+            + "        <xop:Include xmlns:xop=\"http://www.w3.org/2004/08/xop/include\" href=\"ref-to-attachment%40some.domain.org\"/>\r\n"
+            + "      </ns1:Attachment>\r\n"
+            + "    </ns1:Test>\r\n"
+            + "  </soap:Body>\r\n"
+            + "</soap:Envelope>\r\n";
+
+    stubFor(
+        post("/multipart-related")
+            .withMultipartRequestBody(
+                aMultipart()
+                    .withHeader(
+                        "content-type",
+                        equalTo("application/xop+xml; type=\"application/soap+xml\""))
+                    .withBody(equalTo(soapBody)))
+            .withMultipartRequestBody(
+                aMultipart()
+                    .withHeader("content-type", equalTo("text/plain"))
+                    .withHeader("content-id", equalTo("<ref-to-attachment@some.domain.org>"))
+                    .withBody(equalTo("some text/plain content")))
+            .willReturn(ok()));
+
+    final URL url = new URL(wireMockServer.baseUrl() + "/multipart-related");
+    final String boundary = "uuid:" + UUID.randomUUID();
+
+    final HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
Using the old skool HTTP client like this is a bit horrible.

Any reason we can't use the Apache client's support for multipart requests like the other test cases?

Reply:
You're right. I have written the Test with pure `UrlConnection` because I thought that my Knowledge of HttpClient5 is too less to use it with `multipart/related`. I have changed the Code now